### ISSUE TITLE
Animation paths that aren't present are exported

### ISSUE THREAD

**Issue by Suicolen** [KhronosGroup/glTF-Blender-IO#1492]
When creating a simple animation in blender that only involves rotations, scale and translation paths are exported as well.
Steps to reproduce:
1) Set the keyframe time(15 in my case)
2) Rotate the bone using the rotation gizmo 
3) Press 'I' and select Rotation 
4) Set the keyframe time(30 in my case) 
5) Rotate the bone using the rotation gizmo 
6) Press 'I' and select rotation
7) Export as glTF embedded

The animation only contains rotations yet translation and scale paths are exported as well for each target.

I could not include the glTF file as a file so heres the json: https://pastie.io/kwwhqt.json

Edit: it seems like disabling "Always sample animations" when exporting fixes that(that way it only exported rotation paths but also a lot less targets) the animation still works as it should tho, still not sure why that worked tho.

Another edit: the values don't look right either.
Gif showing the current rotation at each keyframe: https://imgur.com/TwV9KmE
Output when parsing: 
```
Channel: rotation and Left leg | Bone id: 29
Input(time) 0.20833333 at element 0
Input(time) 0.625 at element 1
Output(rotation) 0.0, 0.0, 0.0, 0.0 at element 0
Output(rotation) -0.97014254, -1.8371522E-9, -7.3486084E-9, 0.24253552 at element 1
Output(rotation) 0.0, 0.0, 0.0, 0.0 at element 2
Output(rotation) 0.0, 0.0, 0.0, 0.0 at element 3
Output(rotation) 1.0, 5.1076907E-16, 7.574772E-9, 6.743028E-8 at element 4
Output(rotation) 0.0, 0.0, 0.0, 0.0 at element 5
```
glTF file(as json) of the animated version: https://pastie.io/kjjgfm.json

Update: Creating an animation with just 1 keyframe (instead of 2 like before), it seems to work:
```
Channel: rotation and Left leg | Bone id: 29
Input(time) 0.41666666 at element 0
Output(rotation) -0.9663514, -1.948426E-9, -7.319892E-9, 0.2572256 at element 0
```
Which is indeed correct(well excluding the y, z component which are extremely small values altho in blender it shows 0.000): https://i.imgur.com/iJhx3fF.png

I also tried to add another keyframe now (at t = 10 the bone is rotated and at t = 20 it has no rotation) and it seems to work well:
```
Channel: rotation and Left leg | Bone id: 29
Input(time) 0.41666666 at element 0
Input(time) 0.45833334 at element 1
Input(time) 0.5 at element 2
Input(time) 0.5416667 at element 3
Input(time) 0.5833333 at element 4
Input(time) 0.625 at element 5
Input(time) 0.6666667 at element 6
Input(time) 0.7083333 at element 7
Input(time) 0.75 at element 8
Input(time) 0.7916667 at element 9
Input(time) 0.8333333 at element 10
Output(rotation) -0.9663514, -1.9484265E-9, -7.3198927E-9, 0.2572256 at element 0
Output(rotation) -0.9681806, -1.8956074E-9, -7.333748E-9, 0.25025254 at element 1
Output(rotation) -0.97290623, -1.7512896E-9, -7.3695428E-9, 0.23120013 at element 2
Output(rotation) -0.97921526, -1.5363454E-9, -7.4173325E-9, 0.2028238 at element 3
Output(rotation) -0.985791, -1.2723838E-9, -7.467143E-9, 0.16797633 at element 4
Output(rotation) -0.9915522, -9.825135E-10, -7.510782E-9, 0.1297085 at element 5
Output(rotation) -0.99582845, -6.9117195E-10, -7.543174E-9, 0.09124638 at element 6
Output(rotation) -0.9984375, -4.232797E-10, -7.562938E-9, 0.055880003 at element 7
Output(rotation) -0.99963987, -2.0327545E-10, -7.572044E-9, 0.026835663 at element 8
Output(rotation) -0.999974, -5.4607152E-11, -7.5745765E-9, 0.007208896 at element 9
Output(rotation) 1.0, 8.980104E-16, 7.574773E-9, 6.743028E-8 at element 10
```
The only thing im not sure about is why blender exports those extremely small values instead of 0
and also what exactly does the 'always sample animations' toggle does.
NOTE: if it matters im using https://github.com/javagl/JglTF to parse glTF files

**Comment by julienduroure** [comment]
Hello,
Can you please test with a more recent version ( 3.0 RC from here : https://builder.blender.org/download/daily/ )
You can use zip file to provide your blend file.
Thanks !

**Comment by Suicolen** [comment]
Sorry, i thought 2.93.6 was the latest so i used that(wasn't aware that blender had daily builds)
This issue is fixed in 3.0 RC(i no longer have to disable 'Always sample animations' for it to only export the affected keyframes) and it exports as intended.
Only thing im not sure about is why im getting values like '7.574773E-9' when it should be 0 i'd assume that this has to do with some floating point representation stuff but then again 0 has an exact representation(afaik)

**Comment by Suicolen** [comment]
Closing this and asking in a new issue instead
